libsolidity/interface/StandardCompiler.h

```diff
@@ -89,6 +88,10 @@ class StandardCompiler
 		bool viaIR = false;
 	};
 
+	/// Configures @p _compilerStack according to the input settings in @p _inputsAndSettings.
+	static void configure(InputsAndSettings const& _inputsAndSettings, CompilerStack& _compilerStack);
```

Comment:
Reposting my comment from https://github.com/ethereum/solidity/pull/11403#discussion_r633572409:

> Why not make it a factory method? Do you actually need to be able to reconfigure an existing instance of `CompilerStack` in LSP?
> ```suggestion
> 	static CompilerStack makeCompilerStack(InputsAndSettings const& _inputsAndSettings, ReadCallback::Callback _readCallback);
> ```
>
> Also, it might be better to have it as a factory method on `CompilerStack` instead. I think it makes more sense than > exposing it as an internal implementation detail of `StandardCompiler`.

Reply:
@cameel this is a bit tricky. I generally agree that configuration is best placed at object construction, but in here in case, I simply moved the function body of that configure() function out of the `compileSolidity` function, which is invoked by `compile()` which may or may NOT contain settings.

I've moved this configuration step out because in the LSP PR we also need it (without code duplication). We could do that in a second step (different PR only taking care of further improving the CompilerStack API) what do you think?